src/store/actions.js

```diff
@@ -153,6 +154,23 @@ async function fetchBlockHeaderByHash({ state, commit }, { blockHash, done, fail
   commit('endRequest', { identifier: 'fetchBlockHeaderByHash' });
 }
 
+async function fetchBucket( { state, commit }, { marketName }) {
+  commit('startRequest', { identifier: 'fetchBucket' });
+  try {
+    let bucket;
+    if (state.tradeHistory[marketName] && state.tradeHistory[marketName].apiBuckets && state.tradeHistory[marketName].marketName === marketName) {
```

Comment:
If this stops refreshing when the market hasn’t changed, how will new missed trades get shown if websocket disconnects and reconnects?

Reply:
This ought to update if websocket disconnects now. On reOpen it should dispatch to 'fetchMarkets' which in turn dispatches to 'fetchTradeHistory' which gets the api buckets.